standalone-metastore/metastore-catalog/src/test/java/org/apache/iceberg/rest/TestHMSCatalog.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+import org.apache.hadoop.hive.metastore.api.Database;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.Transaction;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.rest.requests.CreateTableRequest;
+import org.apache.iceberg.rest.requests.RenameTableRequest;
+import org.apache.iceberg.types.Types;
+import static org.apache.iceberg.types.Types.NestedField.required;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TestHMSCatalog extends HMSTestBase {
+  public TestHMSCatalog() {
+    super();
+  }
+  
+  @Before
+  @Override
+  public void setUp() throws Exception {
+      super.setUp();
+  }
+  
+  @After
+  @Override
+  public void tearDown() throws Exception {
+      super.tearDown();
+  }
+
+  @Test
+  public void testCreateNamespaceHttp() throws Exception {
+    String ns = "nstesthttp";
+    // list namespaces
+    URL url = new URL("http://hive@localhost:" + catalogPort + "/"+catalogPath+"/v1/namespaces");
+    String jwt = generateJWT();
+    // check namespaces list (ie 0)
+    Object response = clientCall(jwt, url, "GET", null);
+    Assert.assertTrue(response instanceof Map);
+    Map<String, Object> nsrep = (Map<String, Object>) response;
+    List<List<String>> nslist = (List<List<String>>) nsrep.get("namespaces");
```

Comment:
Why is the type List<List<String>> ?
According to Iceberg I can see this:
public List<Namespace> listNamespaces(Namespace ns)

here:
https://github.com/apache/iceberg/blob/31e9dc768cef47df5eb007577f25ef3bbff86f61/core/src/main/java/org/apache/iceberg/rest/RESTCatalog.java#L227

Reply:
This is what comes back from the clientCall; need to check whether this is an unwarranted clientCall bug or something more serious.

Reply:
This result is the json coming back and re-serialized as an object without type. It turns out that a ListNamespaceResponse is a list of Namespace and a Namespace an array of levels thus the double list. Result is normal.